src/Reflex/Class.hs

```diff
@@ -821,6 +825,63 @@ zipListWithEvent f l e = do
   mapAccumMaybe_ f' l e
 
 --------------------------------------------------------------------------------
+-- Sampleable
+--------------------------------------------------------------------------------
+
+-- | A 'Sampleable' type carries a function that can be applied to an 'Event' whenever the 'Event' fires.
+--
+-- 'Sampleable' is intended to be used via the '<@>' and '<@' operators and in conjunction with the `Applicative` interface.
+--
+-- If we have:
+--
+-- > f  :: a -> b -> c -> d
+-- > g  :: a -> b -> d
+-- > b1 :: Behavior t a
+-- > b2 :: Behavior t b
+-- > e  :: Event t c
+--
+-- and an 'Sampleable' instance for `Behavior`, then we get:
+--
+-- > f <$> b1 <*> b2 <@> e :: Event t d
+-- > g <$> b1 <*> b2 <@  e :: Event t d
+--
+-- It is a handy way to combine several 'Behavior's or 'Dynamic's together so that you can sample them all at the times when an `Event` is firing.
+--
+-- The 'Sampleable' class works via the 'attachApplyWithMaybeM' function so that we can implement a wider range of functions with good performance.
+-- Once you have the instance that you need, you should probably be using it via the '<@>' and '<@' operators.
+class Reflex t => Sampleable t f where
+  attachApplyWithMaybeM :: (a -> b -> PushM t (Maybe c)) -> f a -> Event t b -> Event t c
+
+instance Reflex t => Sampleable t Identity where
+  attachApplyWithMaybeM f (Identity a) = push $ \x -> f a x
+
+instance Reflex t => Sampleable t Maybe where
+  attachApplyWithMaybeM f (Just a) = push $ \x -> f a x
+  attachApplyWithMaybeM _ Nothing  = const never
```

Comment:
I can't think of many cases where I'd use the `Sampleable` class with the `Identity` or `Maybe` instances.  I generally just try to create instances for common types where there's only really one valid instance so that people don't have to create orphans or wrap things in newtypes.

I'm happy with getting rid of this if you don't want it, and if it's likely to interfere with future changes then it should definitely go.